testing/test_conftest.py

```diff
@@ -195,7 +195,21 @@ def pytest_addoption(parser):
     reason="symlink not available on this platform",
 )
 def test_conftest_symlink(testdir):
-    """Ensure that conftest.py is used for resolved symlinks."""
+    """
```

Comment:
This test changed the outcome significantly, so, I'd like to elaborate a bit...

What it did was that if the user had the structure:

```
/real
/real/conftest.py
/real/app
/real/app/tests
/real/app/tests/test_foo.py
```

with the following links:

```
/symlinktests -> /real/app/tests (running at symlinktests should fail)
/symlink -> /real (running at /symlink should work)
```

Running a test in `/symlinktests` would still find the `conftest`. This was surprising for me, so, this will probably break someone that expected to have that working...

I still think that the change does make sense and `pytest` should respect the current dir without resolving `symlinks`, which is more flexible (i.e.: users can build their own structure with symlinks)... the other approach required users to build their structure without any symlinks but allowed the creation of symlinks to any folder and the user could enter the symlink to run the tests against the non-symlinked version (which strikes me as odd).

Reply:
This is from me.. :)  (via 7268462b3 #4108)

In this case I've used a symlink "to build my own structure" (to have a shorter path mostly).
`tests -> src/app/tests/`, but with `./src/app/conftest.py` (like you've described).

The workaround/fix here is just moving (or also symlinking (not sure though how that work with your changes)) the `conftest.py`, so that would be OK with me (and maybe there is nobody else expecting this to work then.. :))

And while I like the change in general (less resolving), I'd still like to investigate if it could be made to still work - given that it is below "rootdir" after all, and it is rather surprising that tests might behave differently based on where they are linked from to - although that might also be seen as a feature then.
In general though symlink resolution works as a first step, and then config files (e.g. `.ignore` with ripgrep) are looked up from there (up to a certain root).
In pytest's case I consider `conftest.py` to be a config file, and `rootdir` where it should stop looking then.